.github/workflows/windows.yml

```diff
@@ -136,11 +143,44 @@ jobs:
 
     - name: Enable runtime 32bit size/length/offset limit
       if: matrix.target-arch.name == 'Win32' # Note: this is not a mandatory limitation in any context, we add this just to have at least one configuration testing that feature
+      run: echo "SPARROW_CHECKS='$SPARROW_CHECKS -DSPARROW_ENABLE_32BIT_SIZE_LIMIT=ON'" >> $GITHUB_ENV
+
+    - name: Set CMake generator environment variable
       run: |
-        echo "SPARROW_CHECKS='$SPARROW_CHECKS -DSPARROW_ENABLE_32BIT_SIZE_LIMIT=ON'" >> $GITHUB_ENV
+        echo "CMAKE_GENERATOR=${{ matrix.build-system.name }}" >> $GITHUB_ENV
+
+    - name: Configure nanoarrow
+      working-directory: dependencies/tests/nanoarrow
+      run: |
+        cmake -S ./  \
+          -B ./build \
+          $GENERATOR_EXTRA_FLAGS \
+          -DCMAKE_C_COMPILER_LAUNCHER=ccache \
+          -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
+          -DCMAKE_BUILD_TYPE:STRING=Release \
+          -DCMAKE_PREFIX_PATH=$SPARROW_DEPS_PREFIX \
+          -DCMAKE_INSTALL_PREFIX=$SPARROW_INSTALL_PREFIX \
+          -DCMAKE_COMPILE_WARNING_AS_ERROR=FALSE
+
+    - name: Build nanoarrow
+      working-directory: dependencies/tests/nanoarrow/build
+      run: cmake --build . --config Release --target install
 
     - name: Configure using CMake
-      run: cmake -S ./ -B ./build -DCMAKE_BUILD_TYPE:STRING=${{matrix.config.name}} -DCMAKE_PREFIX_PATH=$SPARROW_DEPS_PREFIX -DCMAKE_INSTALL_PREFIX=$SPARROW_INSTALL_PREFIX -DBUILD_TESTS=ON -DBUILD_EXAMPLES=ON -DUSE_DATE_POLYFILL=${{matrix.toolchain.date-polyfill}} -G "${{matrix.build-system.name}}" $GENERATOR_EXTRA_FLAGS $SPARROW_CHECKS $SPARROW_ADDITIONAL_OPTIONS -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache
+      run: |
+        cmake -S ./ -B ./build \
+        -DCMAKE_BUILD_TYPE:STRING=${{matrix.config.name}} \
+        -DCMAKE_PREFIX_PATH=$SPARROW_DEPS_PREFIX \
+        -DCMAKE_INSTALL_PREFIX=$SPARROW_INSTALL_PREFIX \
+        -DBUILD_TESTS=ON \
+        -DBUILD_EXAMPLES=ON \
+        -DUSE_DATE_POLYFILL=${{matrix.toolchain.date-polyfill}} \
+        $GENERATOR_EXTRA_FLAGS \
+        $SPARROW_CHECKS \
+        $SPARROW_ADDITIONAL_OPTIONS \
+        -DCMAKE_C_COMPILER_LAUNCHER=ccache \
+        -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
+        -DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreadedDLL
```

Comment:
We compile nanoarrow in release so we have to explicitly define CMAKE_MSVC_RUNTIME_LIBRARY to avoid binary mismatch